internal/commands/parameters.go

```diff
@@ -101,22 +108,25 @@ func mergeBundleParameters(installation *store.Installation, ops ...mergeBundleO
 			return err
 		}
 	}
-	if err := matchAndMergeParametersDefinition(installation.Parameters, cfg.params, bndl.Definitions, cfg.strictMode, cfg.stderr); err != nil {
+	if err := matchAndMergeParametersDefinition(installation.Parameters, cfg); err != nil {
 		return err
 	}
 	var err error
-	installation.Parameters, err = bundle.ValuesOrDefaults(installation.Parameters, bndl)
+	installation.Parameters, err = bundle.ValuesOrDefaults(installation.Parameters, installation.Bundle)
 	return err
 }
 
-func matchAndMergeParametersDefinition(currentValues map[string]interface{}, parameterValues map[string]string, parameterDefinitions definition.Definitions, strictMode bool, stderr io.Writer) error {
-	for k, v := range parameterValues {
-		definition, ok := parameterDefinitions[k]
+func matchAndMergeParametersDefinition(currentValues map[string]interface{}, cfg *mergeBundleConfig) error {
+	for k, v := range cfg.params {
+		if cfg.verifyImmutable && cfg.bundle.Parameters.Fields[k].Immutable {
+			return fmt.Errorf("parameter %q is immutable and cannot be overridden", k)
```

Comment:
Shouldn't we use cfg.stderr here too?

Reply:
Here `cfg.stderr` is only used when we want to print a warning message. Here we want to fail.

Reply:
Oh right, sorry